web/shared/src/typeahead.ts

```diff
@@ -35,6 +35,8 @@ type Emoji =
           emoji_name: string;
           reaction_type: "realm_emoji" | "zulip_extra_emoji";
           is_realm_emoji: true;
+          emoji_url?: string;
+          emoji_code: undefined;
```

Comment:
Is `undefined` correct here? I'd expect a realm emoji to have the ID of the `RealmEmoji` object as its ID, for the `realm_emoji` class.

Reply:
Here's one that's undefined, printed from here

```js
export function get_emoji_matcher(query: string): (emoji: EmojiSuggestion) => boolean {
    // replace spaces with underscores for emoji matching
    query = query.replace(/ /g, "_");
    query = clean_query_lowercase(query);

    return function (emoji) {
        console.log(emoji)
        const matches_emoji_literal =
            emoji.reaction_type === "unicode_emoji" &&
            parse_unicode_emoji_code(emoji.emoji_code) === query;
        return matches_emoji_literal || query_matches_string_in_order(query, emoji.emoji_name, "_");
    };
}
```

<img width="764" alt="image" src="https://github.com/zulip/zulip/assets/5634097/339dd8e8-4222-4aae-a504-8c4da3b2cfe4">

And I double checked that it's undefined on `main` too, but less explicitly so:

<img width="636" alt="image" src="https://github.com/zulip/zulip/assets/5634097/c87584e3-b58a-49b4-83b1-3b59a35b314d">